src/subsystems/php/discoverers/default/default.nix

```diff
@@ -0,0 +1,31 @@
+{
+  dlib,
+  lib,
+  subsystem,
+  ...
+}: let
+  l = lib // builtins;
+
+  # get translators for the project
+  getTranslators = path: let
+    nodes = l.readDir path;
```

Comment:
call to readDir could can be prevented by accessing `tree`.
Example: `tree.files ? "composer.lock"`

Reply:
done